lib/experimental/blocks.php

```diff
@@ -79,90 +79,81 @@ function wp_enqueue_block_view_script( $block_name, $args ) {
 }
 
 
+require_once __DIR__ . '/block-bindings/index.php';
 
+if ( ! function_exists( 'gutenberg_process_block_bindings' ) ) {
+	/**
+	 * Process the block bindings attribute.
+	 *
+	 * @param string   $block_content Block Content.
+	 * @param array    $block Block attributes.
+	 * @param WP_Block $block_instance The block instance.
+	 */
+	function gutenberg_process_block_bindings( $block_content, $block, $block_instance ) {
+
+		// Allowed blocks that support block bindings.
+		// TODO: Look for a mechanism to opt-in for this. Maybe adding a property to block attributes?
+		$allowed_blocks = array(
+			'core/paragraph' => array( 'content' ),
+			'core/heading'   => array( 'content' ),
+			'core/image'     => array( 'url', 'title', 'alt' ),
+			'core/button'    => array( 'url', 'text' ),
+		);
 
-$gutenberg_experiments = get_option( 'gutenberg-experiments' );
-if ( $gutenberg_experiments && (
-	array_key_exists( 'gutenberg-block-bindings', $gutenberg_experiments ) ||
-	array_key_exists( 'gutenberg-pattern-partial-syncing', $gutenberg_experiments )
```

Comment:
@kevin940726 Just FYI, I'm removing this check because I think that it's not necessary. The "pattern" source is not loaded unless the `gutenberg-pattern-partial-syncing` experiment is enabled so I think that we don't have to check it here either.

Reply:
Thanks for working on this! We have a PR to remove the experiment for pattern overrides here - [Remove pattern override experiment completely](https://github.com/WordPress/gutenberg/pull/58105#top)

It's close but we need to update a few end to end tests to account for the change in the pattern block.

Hopefully we can merge both PRs without too many conflicts.

Reply:
Great! I think that this one is be ready and we'd like to merge it before the Beta tomorrow. Do you think https://github.com/WordPress/gutenberg/pull/58105 would be merged by then?